GTA does not allow revisit of define types

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
	"fmt"

	"github.com/traefik/yaegi/_test/method38"
)

func main() {
	fmt.Println(method38.Get())
}

```

`method38/a.go`
```go
package method38

import "sync"

func NewPool() Pool { return Pool{} }

type Buffer struct {
	bs   []byte
	pool Pool
}

type Pool struct {
	p *sync.Pool
}

var (
	_pool = NewPool()
	Get   = _pool.Get
)

```

`method38/b.go`




### Expected result

```console
$  go run .sample.go
&{[] {<nil>}}
```
`method38/b.go`
```go
package method38

func (p Pool) Get() *Buffer { return &Buffer{} }

```

### Got

```console
$ yaegi run ./sample.go
run: ./sample.go:6:2: import "github.com/traefik/yaegi/_test/method38" error: github.com/traefik/yaegi/_test/method38/a.go:12:11: undefined selector Get
```


### Yaegi Version

b591ba0e78db7557b5d4e06cc42970d702ce0728

### Additional Notes

_No response_

--- a/_test/method38.go
+++ b/_test/method38.go
@@ -1,0 +1,14 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/traefik/yaegi/_test/method38"
+)
+
+func main() {
+	fmt.Println(method38.Get())
+}
+
+// Output:
+// &{[] {<nil>}}

--- a/_test/method38/a.go
+++ b/_test/method38/a.go
@@ -1,0 +1,19 @@
+package method38
+
+import "sync"
+
+func NewPool() Pool { return Pool{} }
+
+type Buffer struct {
+	bs   []byte
+	pool Pool
+}
+
+type Pool struct {
+	p *sync.Pool
+}
+
+var (
+	_pool = NewPool()
+	Get   = _pool.Get
+)

--- a/_test/method38/b.go
+++ b/_test/method38/b.go
@@ -1,0 +1,3 @@
+package method38
+
+func (p Pool) Get() *Buffer { return &Buffer{} }

--- a/_test/method39.go
+++ b/_test/method39.go
@@ -1,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/traefik/yaegi/_test/method38"
+)
+
+func NewPool() Pool { return Pool{} }
+
+type Buffer struct {
+	bs   []byte
+	pool Pool
+}
+
+type Pool struct {
+	p *sync.Pool
+}
+
+var (
+	_pool = NewPool()
+	Get   = _pool.Get
+)
+
+
+func main() {
+	fmt.Println(Get())
+}
+
+// Error:
+// 17:11: undefined selector Get

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -37,10 +37,17 @@ func (interp *Interpreter) gta(root *node, rpath, importPath, pkgName string) ([
 			}
 
 		case defineStmt:
-			var atyp *itype
+			var (
+				atyp *itype
+				err2 error
+			)
 			if n.nleft+n.nright < len(n.child) {
 				// Type is declared explicitly in the assign expression.
-				if atyp, err = nodeType(interp, sc, n.child[n.nleft]); err != nil {
+				if atyp, err2 = nodeType(interp, sc, n.child[n.nleft]); err2 != nil {
+					// The type does not exist yet, stash the error and come back
+					// when the type is known.
+					n.meta = err2
+					revisit = append(revisit, n)
 					return false
 				}
 			}
@@ -63,7 +70,11 @@ func (interp *Interpreter) gta(root *node, rpath, importPath, pkgName string) ([
 				}
 				typ := atyp
 				if typ == nil {
-					if typ, err = nodeType(interp, sc, src); err != nil || typ == nil {
+					if typ, err2 = nodeType(interp, sc, src); err2 != nil || typ == nil {
+						// The type does is not known yet, stash the error and come back
+						// when the type is known.
+						n.meta = err2
+						revisit = append(revisit, n)
 						return false
 					}
 					val = src.rval
@@ -334,8 +345,13 @@ func (interp *Interpreter) gtaRetry(nodes []*node, importPath, pkgName string) e
 
 	if len(revisit) > 0 {
 		n := revisit[0]
-		if n.kind == typeSpec {
+		switch n.kind {
+		case typeSpec:
 			if err := definedType(n.typ); err != nil {
+				return err
+			}
+		case defineStmt:
+			if err, ok := n.meta.(error); ok {
 				return err
 			}
 		}

--- a/interp/interp.go
+++ b/interp/interp.go
@@ -53,6 +53,7 @@ type node struct {
 	val    interface{}    // static generic value (CFG execution)
 	rval   reflect.Value  // reflection value to let runtime access interpreter (CFG)
 	ident  string         // set if node is a var or func
+	meta   interface{}    // meta stores meta information between gta runs, like errors
 }
 
 func (n *node) shouldBreak() bool {

--- a/interp/interp_consistent_test.go
+++ b/interp/interp_consistent_test.go
@@ -56,6 +56,7 @@ func TestInterpConsistencyBuild(t *testing.T) {
 			file.Name() == "op9.go" || // expect error
 			file.Name() == "bltn0.go" || // expect error
 			file.Name() == "method16.go" || // private struct field
+			file.Name() == "method39.go" || // expect error
 			file.Name() == "switch8.go" || // expect error
 			file.Name() == "switch9.go" || // expect error
 			file.Name() == "switch13.go" || // expect error

